aggregation/main.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// Podcast represents the schema for the "Podcasts" collection
+type Podcast struct {
+	ID     primitive.ObjectID `bson:"_id,omitempty"`
+	Title  string             `bson:"title,omitempty"`
+	Author string             `bson:"author,omitempty"`
+	Tags   []string           `bson:"tags,omitempty"`
+}
+
+// Episode represents the schema for the "Episodes" collection
+type Episode struct {
+	ID          primitive.ObjectID `bson:"_id,omitempty"`
+	Podcast     primitive.ObjectID `bson:"podcast,omitempty"`
+	Title       string             `bson:"title,omitempty"`
+	Description string             `bson:"description,omitempty"`
+	Duration    int32              `bson:"duration,omitempty"`
+}
+
+// PodcastEpisode represents an aggregation result-set for two collections
+type PodcastEpisode struct {
+	ID          primitive.ObjectID `bson:"_id,omitempty"`
+	Podcast     Podcast            `bson:"podcast,omitempty"`
+	Title       string             `bson:"title,omitempty"`
+	Description string             `bson:"description,omitempty"`
+	Duration    int32              `bson:"duration,omitempty"`
+	PublishDate int32              `bson:"publish_date,omitempty"`
+}
+
+func main() {
+	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(os.Getenv("ATLAS_URI")))
+	if err != nil {
+		panic(err)
+	}
+	defer client.Disconnect(ctx)
+
+	database := client.Database("quickstart")
+	episodesCollection := database.Collection("episodes")
+
+	id, _ := primitive.ObjectIDFromHex("5e3b37e51c9d4400004117e6")
+
+	showInfoCursor, err := episodesCollection.Aggregate(ctx, []bson.D{
+		bson.D{{"$match", bson.D{{"podcast", id}}}},
+		bson.D{{"$group", bson.D{{"_id", "$podcast"}, {"total", bson.D{{"$sum", "$duration"}}}}}},
+	})
+	if err != nil {
+		panic(err)
+	}
+	var showsWithInfo []interface{}
```

Comment:
I generally avoid decoding into `interface{}`. It yields a `bson.D` so I find it much clearer to just write `bson.D` instead. Also, this is just being used to print, so I'd recommend using `[]bson.M` here, which is much more readable when printed. A `bson.D` printed out looks strange because it's actually a slice of structs so you see all sorts of irrelevant info.